src/librustdoc/clean/types.rs

```diff
@@ -1304,6 +1302,49 @@ crate enum TypeKind {
     TraitAlias,
 }
 
+impl<'a> From<&'a hir::def::DefKind> for TypeKind {
+    fn from(other: &hir::def::DefKind) -> Self {
```

Comment:
Do you want to add

```suggestion
    fn from(other: &'a hir::def::DefKind) -> Self {
```

to make it explicit that it's the same lifetime?

Reply:
They're not supposed to have the same lifetime. ;)

Reply:
What do you mean? They're supposed to be the exact same type.

Reply:
Oh, yes sorry. I wanted to say that the compiler didn't allow to elide lifetimes in the `impl` definition but that there was not much point into adding it into the `from` definition. (with complete explanations, I guess it makes more sense haha)

Reply:
I feel unclear: do you plan to add the lifetime or not?

Reply:
Why is this implementation even here? `DefKind` is a small `Copy` type, there's no reason to pass it by reference.

Reply:
It's completely uneeded, you're both right. It came from the modification of `impl Clean<TypeKind> for hir::def::DefKind {`. It'll appear more clearly in the last commit I'll push.